Modification to issue #1516

MohamedLebda opened the pull request.

hboyd2003 committed:
Replace es_419 locale using es

Commit:
Merge pull request #1455 from hboyd2003/master

Revert removal of locales and updates es_419 with es

adokseo committed:
Updated locales

adokseo committed:
Updated locales

adokseo committed:
Updated content-scripts, etc.

adokseo committed:
Removed Google Analytics

adokseo committed:
Some updates

Commit:
Merge https://github.com/hboyd2003/Youtube-Extension

Commit:
Merge pull request #1462 from code-for-charity/revert-1458-master

Revert "Cleanup & rework README.md"

Commit:
Create firefox-addon-store-builder.py

Commit:
a

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

kaplantm committed:
add hide comments count option (issue #1440)

kaplantm committed:
add hide comments count option locales (issue #1440)

Commit:
Merge pull request #1464 from kaplantm/issue-1440-hide-comments-count

Issue 1440 hide comments count

dodieboy committed:
Fixed player size not working

dodieboy committed:
Fix locale.py codec error

dodieboy committed:
Disable debug command

dodieboy committed:
Add hide detail button and button text

Commit:
Merge pull request #1465 from dodieboy/Fix

Port over my feature from v3 to v4

Commit:
Create CONTRIBUTING.md

PeterDaveHello committed:
Update zh_TW translation

dodieboy committed:
Fixed hide detail button text and hide like and dislike button not working

dodieboy committed:
Improved detail button size when text is hidden

dodieboy committed:
Update channelVideosCount()

Port commit fd4f5d9 by @hboyd2003 to v4

dodieboy committed:
Update playerPlaybackSpeed()

Port commit 4fc9265by @hboyd2003 to v4

dodieboy committed:
Update playerControls()

Port commit 484fd34 by @JuniorGenius to v4

dodieboy committed:
Update subtitles()

Port 6ae0483 by @MyBestFriend to v4

dodieboy committed:
Update shortcutStop()

Port commit 27d848b by @JuniorGenius to v4

dodieboy committed:
Update channelDefaultTab()

dodieboy committed:
Update mini_player__setSize()

Port commit 9401b84 by @samit43 to v4

dodieboy committed:
Update Header position

dodieboy committed:
Update playlistRepeat()

dodieboy committed:
Improve autoplay()

dodieboy committed:
Improved sidebar

dodieboy committed:
Fixed forced theater mode not working

dodieboy committed:
Update player control option

fix hide autoplay not working
move player control option from player to appearance player

dodieboy committed:
Update playlistUpNextAutoplay()

Port commit a28e712 by @MyBestFriend to v4

dodieboy committed:
Update alwaysShowProgressBar()

Port commit ead30b1 from @BitbenderM to v4
and more

dodieboy committed:
Remove description "classic" option

it cause the whole info bar (title of video) to be missing

dodieboy committed:
Add day of week (#1379)

Commit:
Merge pull request #1466 from PeterDaveHello/patch-1

Update zh_TW translation

dodieboy committed:
Revert "Update playerPlaybackSpeed()"

This reverts commit 539e413c46c93b2be841c3d6f30f92c169ee6751.

dodieboy committed:
MutationObserver code optimization

dodieboy committed:
Remove unused code

FedericoSlongo committed:
Update messages.json

dodieboy committed:
Fix channelDefaultTab() console error

dodieboy committed:
Fix view count and date disappear when too many things in detail

dodieboy committed:
Fix dayOfWeek() not working for steaming and other language

FabrisNick committed:
Better translation

FedericoSlongo committed:
Merge pull request #1 from CoachYT1/patch-1

Better cpu

dodieboy committed:
Improved playlistRepeat()

dodieboy committed:
Fixed pageOnFocus() error

Fixed "ImprovedTube.playerAutopauseWhenSwitchingTabs is not a function" error

dodieboy committed:
Bump version

dodieboy committed:
Improved hover header

Fix #1369

dodieboy committed:
Updated duration_with_speed()

it will only show when show-remaining-duration is enabled

dodieboy committed:
Improved hide detail button and button text

now used the old design

dodieboy committed:
Merge branch 'master' into fix

dodieboy committed:
Updated playerSize()

Commit:
Merge pull request #1469 from dodieboy/fix

Commit:
Merge pull request #1472 from FedericoSlongo/master

Commit:
Update README.md

dodieboy committed:
Format satus code

dodieboy committed:
Fixed H.264's alert/dialogue broken

dodieboy committed:
Updated improvedtubeYoutubeIcon()

Bring back improvedtubeYoutubeIcon() and updated it to link with the new yt header code

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Merge pull request #1478 from dodieboy/Fix

dodieboy committed:
Fix expand description not working

Commit:
Merge pull request #1484 from dodieboy/Fix

dodieboy committed:
Fixed theme not working

dodieboy committed:
Fixed theme not working

Commit:
Merge pull request #1488 from dodieboy/Theme-Fix

Fix https://github.com/code-for-charity/ImprovedTube-for-YouTube/issues/1474

0x1-1 committed:
Update for missing translations

Update for missing translations

Commit:
Merge pull request #1489 from daswareinfach/master

Commit:
Update messages.json

Commit:
Update messages.json

Commit:
Update player.js

Commit:
Update player.js

Commit:
name to support old user setting

Commit:
Update details.css

Commit:
Update background.js

Commit:
Update background.js

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update and rename CONTRIBUTING.md to CONTRIBUTINGmd

Commit:
Update and rename CONTRIBUTINGmd to CONTRIBUTING.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

noland96 committed:
General' Features don't appear in 'Search' & Settings: 'Activate Features

noland96 committed:
delete .idea/

ImprovedTube committed:
Merge pull request #1510 from vanquan805/fix-1502

Fix #1502: 'General' Features don't appear in 'Search' & Settings: 'Activate Features'

noland96 committed:
Fix 1501: The setting of channel 'default channel tab' feature isn't working after YTB changed the layout.

ImprovedTube committed:
Merge pull request #1512 from vanquan805/fix-1501

Fix 1501: The setting of channel 'default channel tab' feature isn't …

ImprovedTube committed:
Delete .gitignore

dodieboy committed:
Fix comments "collapsed" break short comment

dodieboy committed:
Fixed skip overlay not working

dodieboy committed:
Update channel regex

dodieboy committed:
Fix recently uploaded and popular buttons missing on channel when header on hover

ImprovedTube committed:
Merge pull request #1515 from dodieboy/Fix

kenclaron committed:
Added Russian localization

ImprovedTube committed:
Update README.md

ImprovedTube committed:
Merge pull request #1518 from kenclaron/feature/ru-locale

Added Russian localization

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

ImprovedTube committed:
Update README.md

ImprovedTube committed:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

ImprovedTube committed:
Update SECURITY.md

Commit:
Update README.md

FedericoSlongo committed:
Updated italian translation

I've translated some more lines to Italian

ImprovedTube committed:
Merge pull request #1533 from FedericoSlongo/patch-2

samyk committed:
fix bug where search isn't focused (dupe ids)

samyk committed:
allow onhover header to show when search focused

samyk committed:
Revert "allow onhover header to show when search focused"

This reverts commit cb2eeb6906fef23b62427b5f8511bb35e88857a2.

samyk committed:
allow onhover header to show when search focused

samyk committed:
fix like button shortcut

samyk committed:
fix dislike shortcut

samyk committed:
Merge branches 'fix-search-when-header-hidden' and 'fix-like-shortcut' of github.com:samyk/ImprovedTube-for-YouTube into samy

ImprovedTube committed:
Merge pull request #1537 from samyk/fix-like-shortcut

Resolves https://github.com/code-for-charity/ImprovedTube-for-YouTube/issues/1525

ImprovedTube committed:
Merge pull request #1535 from samyk/fix-search-hotkey

Resolves https://github.com/code-for-charity/ImprovedTube-for-YouTube/issues/1534

ImprovedTube committed:
Merge pull request #1536 from samyk/fix-search-when-header-hidden

Resolves https://github.com/code-for-charity/ImprovedTube-for-YouTube/issues/884

samyk committed:
Add ability to set speed on all vids except music

ImprovedTube committed:
Merge pull request #1539 from samyk/force-speed-but-not-for-music

Resolves https://github.com/code-for-charity/ImprovedTube-for-YouTube/issues/1538

ImprovedTube committed:
Update building for browser stores.md

ImprovedTube committed:
Update building for browser stores.md

ImprovedTube committed:
Update building for browser stores.md

ImprovedTube committed:
Delete firefox-addon-store-builder.py

ImprovedTube committed:
Quick revision

ImprovedTube committed:
Quick revision

ImprovedTube committed:
Quick revision

ImprovedTube committed:
removing typo: get('watched') = {};

ImprovedTube committed:
Update README.md

ImprovedTube committed:
typos

ImprovedTube committed:
Update README.md

ImprovedTube committed:
Update README.md

qlife1146 committed:
Modifying Korean Translation

qlife1146 committed:
Modifying Korean Translation

ImprovedTube committed:
Update README.md

qlife1146 committed:
There was a bug in the video count function. Please check the API key.

qlife1146 committed:
Modify text and add content to the translation file for translation.

ImprovedTube committed:
Merge pull request #1547 from qlife1146/master

ImprovedTube committed:
Update README.md

ImprovedTube committed:
Update LICENSE

ElectroHeavenVN committed:
Update messages.json

MohamedLebda committed:
fixed issue #1516

MohamedLebda committed:
fixed issue #1516

ImprovedTube committed:
Merge pull request #1562 from MohamedLebda/master

dojoVader committed:
Merge pull request #1 from code-for-charity/master

New changes

dojoVader committed:
Bugfix for #1555

MohamedLebda committed:
fixed issue #1553

ImprovedTube committed:
Merge pull request #1571 from dojoVader/bugfix/1555/header-bug

Bugfix/1555/header bug

ImprovedTube committed:
Merge pull request #1559 from ElectroHeavenVN/master

MohamedLebda committed:
modified issue #1553

MohamedLebda committed:
modified issue #1553

ImprovedTube committed:
Merge pull request #1572 from MohamedLebda/master

MohamedLebda committed:
fix issu 1508

MohamedLebda committed:
fixed issu 1508

ImprovedTube committed:
Merge pull request #1585 from MohamedLebda/master

ImprovedTube committed:
expand description. new css

ImprovedTube committed:
'details' are "ytd-watch-metadata" now

Commit:
fix description

MohamedLebda committed:
Modification to fix  issue #1516

dodieboy committed:
Fix custom theme not working

dodieboy committed:
Fix dark mode not working

Now the theme will pick the youtube appearance by itself.

dodieboy committed:
Fix youtube appearance setting is not set correctly when change theme

The page will now reload when need to change appearance

MohamedLebda committed:
Merge remote-tracking branch 'upstream/master'

MohamedLebda committed:
Merge changes from upstream

MohamedLebda committed:
fixed issu #1400

MohamedLebda committed:
fix issue #1400

ImprovedTube committed:
Merge pull request #1587 from MohamedLebda/master

ImprovedTube committed:
Merge pull request #1589 from dodieboy/Theme-Fix

ImprovedTube committed:
Revert "Modification to fix  issue #1516"

ImprovedTube committed:
Merge pull request #1591 from code-for-charity/revert-1587-master

ImprovedTube committed:
Update manifest.json

ImprovedTube committed:
Create manifest2.js

ImprovedTube committed:
Rename manifest2.js to manifest2.json

Commit:
quick remedy #1589 #1592

Commit:
Merge branch 'HEAD' of https://ImprovedTube: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@github.com/ImprovedTube/Youtube-Extension

Commit:
title

ImprovedTube committed:
pageType regex & order optimization (+a remedy for #1456)

ImprovedTube committed:
typo

dodieboy committed:
Fix theme cookie error

Rewrite the whole theme cookie setting code, now it will use ImprovedTube.setCookie()

dodieboy committed:
Fix player element overlap

Port #1372 to V4

ImprovedTube committed:
1. location.pathname is shorter to regex than location.href    
2. new channel url update

ImprovedTube committed:
typo

ImprovedTube committed:
Merge branch 'master' into test

ImprovedTube committed:
Merge pull request #1596 from code-for-charity/test

dodieboy committed:
Fixed hideDetailButton() error & update hide detail button CSS

dodieboy committed:
Fix ImprovedTube.description() error

ImprovedTube committed:
fix

ImprovedTube committed:
Merge pull request #1597 from dodieboy/Fix

ImprovedTube committed:
typo

MohamedLebda committed:
Modification to issue #1516

MohamedLebda committed:
issue #1552

ImprovedTube commented:
>  || ( ImprovedTube.storage.only_one_player_instance_playing && ImprovedTube.storage.newTab)

why disabling autoplay for any new tab? (=even if no other tab is playing already?)

ImprovedTube committed:
uncommenting #1589 darkmode cookie

ImprovedTube committed:
Merge pull request #1603 from MohamedLebda/issue-1552

thanks

ImprovedTube committed:
#1594 #1590

MohamedLebda commented:
> why storage?

so I can put that condition
content-scripts/website-context/youtube-features/player.js
    ) || ( ImprovedTube.storage.only_one_player_instance_playing && ImprovedTube.storage.newTab)

 and its purpose is to stop any new tap from auto-starting videos whenever you change the tab to it, only when you open the only one player instance feature at the same time it only stops the new taps videos meaning that if you didn't open a new tap the background won't send the new tap opened massage so the second part of the condition will be false even tho the one player instance is still true which means that it will not stop any new videos from auto-starting in the same page you are actively in and the same goes if you open a new tap so it will only affect the start of the video of the new tap so that the user whenever he opens the only player instance can browse and open new tabs without the original video stops only when he manually starts a video in the new tap the original tap stop so no two videos play in the same time

dodieboy committed:
Fix header theme not correct

dodieboy committed:
Improved ImprovedTube.themes()

We will let youtube to pick the theme for default theme now

dodieboy committed:
Fix ImprovedTube.description() bug

Prevent "Uncaught TypeError: Cannot read properties of undefined (reading 'id')"

ImprovedTube commented:
> ImprovedTube.storage.only_one_player_instance_playing && ImprovedTube.storage.newTab

permanent storage. 
so it will stop the first tab from playing if that one is opened in the background (ok, since browser will stop it anyways)

just as said ImprovedTube.storage.newTab = false might stay true always
(
> wont happen for some users, who dont navigate on youtube
)
if a user doesnt navigate on youtube but only open & close single videos ImprovedTube.video_url !== location.href


just how it looks to me, didnt test

ImprovedTube committed:
English

MohamedLebda commented:
> just as said ImprovedTube.storage.newTab = false might stay true always

even if it  ImprovedTube.storage.newTab = true because there is no new video started it still won't stop the video from playing or have an effect 
    
    if (ImprovedTube.allow_autoplay === false  &&
    
so let's say that the user opened a new tab (now ImprovedTube.storage.newTab = true and ImprovedTube.storage.only_one_player_instance_playing)  it will stop the new tab from playing the video, not the original tab.
so if the user stopped the original video and didn't open a new video  (now still ImprovedTube.storage.newTab = true and ImprovedTube.storage.only_one_player_instance_playing) then the user tried to start any new video whether the original video or a new tab video( now still ImprovedTube.storage.newTab = true and ImprovedTube.storage.only_one_player_instance_playing) but the first part of the if condition will be false (ImprovedTube.allow_autoplay === true) so it won't have any effect and once he starts a new video whether in the original tab or the new tab then  

     if (ImprovedTube.video_url !== location.href) {
        ImprovedTube.allow_autoplay = false;
		ImprovedTube.video_url = location.href
		ImprovedTube.storage.newTab = false

so it will be false so it won't stop any new video to start from auto-playing in the same tab 

and if the user stops the one player instance before opening any new video in the same tab which means that 
ImprovedTube.storage.newTab will always be true because it's in the storage but it will have no effect because 

        ) || ( ImprovedTube.storage.only_one_player_instance_playing && ImprovedTube.storage.newTab)

MohamedLebda commented:
> renamed the features ~`only one player instance playing`~ `Don't let a second video auto-start at once`
> moved it after ~`autopause when switching tab` ~ `Pause while not visible`
> and we could add a third one: `Pause the first video when a second one auto-starts`

do you mean dividing the feature into 3 new features?

ImprovedTube committed:
English

ImprovedTube committed:
English

ImprovedTube committed:
Import & delete old settings

ImprovedTube committed:
CSS fixes. Tidying options pages.

ImprovedTube committed:
just a label

ImprovedTube committed:
Update player.js

ImprovedTube committed:
Update player.js

ImprovedTube committed:
haha

ImprovedTube commented:
hi @MohamedLebda! 
> `ImprovedTube.allow_autoplay === false  &&`
this is  just an exception to autoplay:off,  true only after keyboard activity.  
(said in your pull request about autoplay:off:)
Thus wont be true if a user just opens the first tab with no activity.

(btw, found a clue why  'ad-showing'  was there?)

>ImprovedTube.storage.newTab
(can be in Ram: https://developer.chrome.com/docs/extensions/reference/storage/#storage-areas )

> do you mean dividing the feature into 3 new features?

no, sorry,  we already have two options of **Auto-pause**: 
<img width="219" alt="autopause toggle" src="https://user-images.githubusercontent.com/25022245/224523488-66433d0a-d7cd-4405-b323-8fecc6bf2720.png">
(just made them disable each other)
1. ...when switching tabs  = <del> background playing </del>        
<i>
2. ...only one instance  = allowing one background player 

+3.  (or 1.1.)  Auto-pause, while video is scrolled away (like our 'custom mini player' will do)

+ 2.1. Allow (first) video to play when tab is opened in background    (since chrome&firefox block autoplay in background)
 +2.1.1.  Keep all background videos in queue.  - But In that case we could rather would be smarter to ignore opening the tabs/close immediately and turn the first tab into a youtube playlist link 
</i>

more importantly:
- we can expand the feature to all standard \<video\> ?
- we can go back to manifest2

ImprovedTube committed:
English

ImprovedTube committed:
Merge pull request #1604 from dodieboy/Fix

ImprovedTube committed:
Version README.md

ImprovedTube committed:
Update README.md

ImprovedTube committed:
Update manifest.json

ImprovedTube committed:
Update manifest2.json

MohamedLebda committed:
new option #1563

ImprovedTube committed:
Merge pull request #1610 from MohamedLebda/new-option-1563

ImprovedTube committed:
#1536

ImprovedTube committed:
#1536

sapondanaisriwan committed:
Fixed ratio bar not showing

ImprovedTube committed:
Merge pull request #1616 from sapondanaisriwan/fixRatioBarNotShow

Fixed ratio bar not showing.

ImprovedTube committed:
Merge branch 'master' of https://github.com/code-for-charity/ImprovedTube-for-YouTube

ImprovedTube committed:
#1607 fix

ImprovedTube committed:
Video Description (+Buttons) update (finally)

ImprovedTube committed:
no right margin, for all screen sizes

ImprovedTube committed:
Themes, [temporary] improvements #1597 #1604 thanks @dodieboy

ImprovedTube committed:
debug

ImprovedTube committed:
(some new colors) & please check https://github.com/code-for-charity/SATUS/

ImprovedTube committed:
quick debug

ImprovedTube committed:
version

ImprovedTube committed:
Update README.md

MohamedLebda committed:
Merge branch 'code-for-charity:master' into master

MohamedLebda committed:
Modification to issue #1516

MohamedLebda commented:
> (thus won't be true if a user opens the first tab with no activity? )

yes but if the user opened the first (video) tab with no activity the improved.storage.newTab = false
 if (ImprovedTube.video_url !== location.href) {
    ImprovedTube.allow_autoplay = false;
	ImprovedTube.video_url = location.href
	ImprovedTube.storage.newTab = false
